File: frontend/src/services/inventory.service.js
import axios from 'axios';
import authHeader from './auth-header';

const API_URL = (process.env.REACT_APP_API_URL || '') + '/api/inventory/';

class InventoryService {
  getAllInventory() {
    return axios.get(API_URL, { headers: authHeader() });
  }

  getInventoryById(id) {
    return axios.get(API_URL + id, { headers: authHeader() });
  }

  createOrUpdateInventory(inventory) {
    return axios.post(API_URL, inventory, { headers: authHeader() });
  }

  deleteInventory(id) {
    return axios.delete(API_URL + id, { headers: authHeader() });
  }

  getInventoryByNode(nodeId) {
    return axios.get(API_URL + 'node/' + nodeId, { headers: authHeader() });
  }

  getInventoryByProduct(productId) {
    return axios.get(API_URL + 'product/' + productId, { headers: authHeader() });
  }

  updateInventory(id, inventory) {
    return axios.put(API_URL + id, inventory, { headers: authHeader() });
  }

  getLowStockInventory() {
    return axios.get(API_URL + 'low-stock', { headers: authHeader() });
  }
}

const inventoryService = new InventoryService();
export default inventoryService;

File: backend/supply-chain-visualizer/src/main/java/com/example/supplychainvisualizer/service/InventoryService.java
package com.example.supplychainvisualizer.service;

import com.example.supplychainvisualizer.dto.InventoryDto;

import java.util.List;
import java.util.Optional;

public interface InventoryService {
    List<InventoryDto> getAllInventory();
    Optional<InventoryDto> getInventoryById(Long id);
    InventoryDto createOrUpdateInventory(InventoryDto inventoryDto);
    Optional<InventoryDto> updateInventoryById(Long id, InventoryDto inventoryDto);
    boolean deleteInventory(Long id);
    List<InventoryDto> getInventoryByNode(Long nodeId);
    List<InventoryDto> getInventoryByProduct(Long productId);
    List<InventoryDto> getLowStockInventory();
}
